Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGridColumnHeadersPresenter.cs

```diff
@@ -243,7 +243,7 @@ private static void EnsureColumnHeaderClip(DataGridColumnHeader columnHeader, do
             }
         }
 
-        private static void EnsureColumnHeaderClipNew(DataGridColumnHeader columnHeader, double width, double height, double frozenLeftEdge, double columnHeaderLeftEdge, double frozenRightEdge)
+        private static void EnsureColumnHeaderClipNew(DataGridColumnHeader columnHeader, double width, double height, double frozenLeftEdge, double columnHeaderLeftEdge, double frozenRightEdge, bool hasRightFrozenColumn)
```

Comment:
Change
`if (columnHeaderLeftEdge + width > frozenRightEdge)` 
in L262 to
`if (hasRightFrozenColumn && columnHeaderLeftEdge + width - frozenLeftEdge > frozenRightEdge)`